src/main/scala-2.12/org/apache/james/gatling/jmap/scenari/JmapBigSetScenario.scala

```diff
@@ -16,12 +16,13 @@ import scala.concurrent.duration.{Duration, DurationInt}
  */
 class JmapBigSetScenario {
 
-  def generate(duration: Duration, numberOfMailboxes: Int, numberOfMessages: Int, userFeeder: UserFeeder, userPicker: UserPicker): ScenarioBuilder = {
+  def generate(duration: Duration, numberOfMailboxes: Int, numberOfMessages: Int, userFeeder: UserFeeder, recipientFeeder: FeederBuilder): ScenarioBuilder = {
     def numberOfMailboxesPerUser: Int = numberOfMailboxes + numberOfSystemMailboxes
 
     scenario("JMAP scenario on multiple mailboxes containing multiple messages")
       .feed(userFeeder)
-      .exec(CommonSteps.provisionUsersWithMailboxesAndMessages(userPicker, numberOfMailboxes, numberOfMessages))
+      .feed(recipientFeeder)
```

Comment:
I find it a bit weird to use a feeder for something that is related to a specific action.
I would rather inject a random user in the session at the right time by doing:
```scala
.exec(session => session.put("recipient", userPicker.pick()))
```

Reply:
for me this is exactly what a feeder is doing. The UserPicker reproduces manually what a feeder is doing (taking a random user from a Seq), but putting it in the session must be done apart.

If you prefer I can use this feeder only in the sendMessages method, as I did for the MailFeeder.

Reply:
I agree that your solution feel simpler.

Reply:
I moved the feeder. At the end it just replaces the UserPicker with proper Gatling meaning. I also removed the UserPicker by the way.